packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -1,75 +1,135 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import { chainPropTypes, integerPropType } from '@material-ui/utils';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { chainPropTypes, integerPropType } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses, isHostComponent } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import InputBase from '../InputBase';
 import MenuItem from '../MenuItem';
 import Select from '../Select';
 import TableCell from '../TableCell';
 import Toolbar from '../Toolbar';
-import Typography from '../Typography';
 import TablePaginationActions from './TablePaginationActions';
 import useId from '../utils/useId';
+import tablePaginationClasses, { getTablePaginationUtilityClass } from './tablePaginationClasses';
+
+const TablePaginationRoot = experimentalStyled(
+  TableCell,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)(({ theme }) => ({
+  overflow: 'auto',
+  color: theme.palette.text.primary,
+  fontSize: theme.typography.pxToRem(14),
+  // Increase the specificity to override TableCell.
+  '&:last-child': {
+    padding: 0,
+  },
+}));
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    color: theme.palette.text.primary,
-    fontSize: theme.typography.pxToRem(14),
-    overflow: 'auto',
-    // Increase the specificity to override TableCell.
-    '&:last-child': {
-      padding: 0,
-    },
+const TablePaginationToolbar = experimentalStyled(
+  Toolbar,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Toolbar',
+    overridesResolver: (props, styles) => styles.toolbar,
   },
-  /* Styles applied to the Toolbar component. */
-  toolbar: {
+)(({ theme }) => ({
+  minHeight: 52,
+  paddingRight: 2,
+  [`${theme.breakpoints.up('xs')} and (orientation: landscape)`]: {
     minHeight: 52,
-    paddingRight: 2,
   },
-  /* Styles applied to the spacer element. */
-  spacer: {
-    flex: '1 1 100%',
+  [theme.breakpoints.up('sm')]: {
```

Comment:
Where are the breakpoints coming from?

Reply:
As mentioned above, it is needed to override default Toolbar styles (which has breakpoint)

> ![image](https://user-images.githubusercontent.com/18292247/115109616-1113e680-9fa1-11eb-98f7-b35d9471b6f7.png)
> 
> Styling is a bit ugly, is this the expected result? (v4 implementation does not work, I think it is because emotion merge the styles, so I need to add breakpoints to overrides)
> 
> ```js
> const TablePaginationToolbar = experimentalStyled(
>   Toolbar,
>   { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
>   {
>     name: 'MuiTablePagination',
>     slot: 'Toolbar',
>     overridesResolver: makeOverridesResolver('toolbar'),
>   },
> )(({ theme }) => ({
>   minHeight: 52,
>   paddingRight: 2,
>   [`${theme.breakpoints.up('xs')} and (orientation: landscape)`]: {
>     minHeight: 52,
>   },
>   [theme.breakpoints.up('sm')]: {
>     minHeight: 52,
>   },
> }));
> ```

Reply:
This is the result without breakpoints. `min-height: 52px` is overwritten.
![image](https://user-images.githubusercontent.com/18292247/115752466-52066380-a3c4-11eb-9c24-c1a81d249d35.png)